scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/synchronization/AOSynchronization.java

```diff
@@ -0,0 +1,862 @@
+                                               value,
+                                               (BiFunction<? super Serializable, ? super Serializable, ? extends Serializable>) evaluateClosure(remappingFunction,
+                                                                                                                                                BiFunction.class));
+            commitIfNeeded(channel);
+            logWithContext(taskid,
+                           channel,
+                           "Run merge" + ON_KEY + QUOTE + key + QUOTE + WITH_VALUE + value + RESULT_IS + answer,
+                           Level.DEBUG);
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+        return answer;
+    }
+
+    @Override
+    public PredicateActionResult conditionalCompute(String originator, TaskId taskid, String channel, String key,
+            String predicate, String thenRemappingFunction)
+            throws InvalidChannelException, CompilationException, IOException {
+        PredicateActionResult answer;
+        try {
+            Channel chosenChannel = getChannel(channel);
+            Serializable value = chosenChannel.get(key);
+
+            if (evaluateClosure(predicate, BiPredicate.class).test(key, value)) {
+                answer = new PredicateActionResult(true,
+                                                   chosenChannel.compute(key,
+                                                                         (BiFunction<? super String, ? super Serializable, ? extends Serializable>) evaluateClosure(thenRemappingFunction,
+                                                                                                                                                                    BiFunction.class)));
+                commitIfNeeded(channel);
+            } else {
+                answer = new PredicateActionResult(false, value);
+            }
+            logWithContext(taskid,
+                           channel,
+                           "Run conditionalCompute" + ON_KEY + QUOTE + key + QUOTE + RESULT_IS + answer,
+                           Level.DEBUG);
+
+        } catch (InvalidChannelException | CompilationException | IOException e) {
+            throw e;
+        } catch (Exception e) {
+            throw new ClosureEvaluationException(EXCEPTION_WHEN_EVALUATING_CLOSURE + StackTraceUtil.getStackTrace(e));
+        }
+        return answer;
+    }
+
+    @Override
+    public PredicateActionResult conditionalCompute(String originator, TaskId taskid, String channel, String key,
+            String predicate, String thenRemappingFunction, String elseRemappingFunction)
+            throws InvalidChannelException, CompilationException, IOException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the redundant '!unknownSymbol!' thrown exception declaration(s). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)